core/src/main/java/org/elasticsearch/search/suggest/SuggestionBuilder.java

```diff
@@ -212,6 +223,8 @@ protected void populateCommonFields(MapperService mapperService,
                 throw new IllegalArgumentException("Analyzer [" + luceneAnalyzer + "] doesn't exists");
             }
             suggestionContext.setAnalyzer(luceneAnalyzer);
+        } else {
+            suggestionContext.setAnalyzer(mapperService.searchAnalyzer());
         }
```

Comment:
@areek @cbuescher I like Areek's suggestion and the reasoning.  I think its worth implementing now in `populateCommonFields` as it is currently confusing to have some of this logic in `PhraseSuggestionBuilder`, with appropriate tests, unless you want to hold off?  In the case of the `completion` suggester, if the analyzer must be defined at mapping time, then what about overriding `setAnalyzer` in the `CompletionSuggestionBuilder` to set the `analyzer` field to `null` no matter what the input?